MangaDex: Clean manga descriptions

curche opened the pull request:
Closes #10610

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

alessandrojean commented:
> Extra Notes usually just has links to external sites which include anime info/licensors/alternative raws; everything on other websites.

That's why I suggested that the current approach of removing them is good.

alessandrojean commented on src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt:
```diff
@@ -75,7 +75,12 @@ class MangaDexHelper() {
         }
-        return Parser.unescapeEntities(intermediate, false)
+        val unescapedString = Parser.unescapeEntities(intermediate, false)
+        val linkRegex = "\\[([^]]+)\\]\\(([^)]+)\\)".toRegex()
```
Maybe make it a constant?

curche replied:
Sure, makes sense to move it to a companion object.

CarlosEsco commented:
Does Tachi allow selecting text in the summary?  If so I dont see why the urls have to be removed

CarlosEsco commented on src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt:
```diff
@@ -75,7 +75,12 @@ class MangaDexHelper() {
         while (bbRegex.containsMatchIn(intermediate)) {
```
i would strip out anything relating to bbcode/bbregex as 99% of manga have been converted to markdown

curche replied:
I had my doubts but kept it for those last edge cases, if any. Will remove

alessandrojean commented:
> Does Tachi allow selecting text in the summary? If so I dont see why the urls have to be removed

It doesn't, but there's an open issue in the main repo requesting this feature, since J2K have.

CarlosEsco commented:
> > Does Tachi allow selecting text in the summary? If so I dont see why the urls have to be removed
> 
> It doesn't, but there's an open issue in the main repo requesting this feature, since J2K have.

Curche can you post a screen with keeping links vs one with out?  at the end of the day they can also click the webview if they really wanted to see links

curche commented:
(Apologies to everyone for whom this doesn't render correctly)

### Opening a random manga from Latest looks fine
| Before | Now |
|--------------|-------------|
| ![Screenshot_20220204-002023_Tachiyomi_swan_2](https://user-images.githubusercontent.com/72807749/152409919-dee72128-8408-4159-af6a-49cb5dbd2ae1.png) | ![Screenshot_20220204-001908_Tachiyomi_swan_1](https://user-images.githubusercontent.com/72807749/152409933-eeaa3344-fa15-4e4a-a5e7-9e9cb4b48953.png) |

### Sometimes the old bbcode cleaning messes up and helps us out
| Before | Now |
|--------------|-------------|
| ![Screenshot_20220204-001408_TachiyomiSY_Mieruko_2](https://user-images.githubusercontent.com/72807749/152409943-93ba7e7b-22e3-4f72-9e83-72c67af452fe.png) | ![Screenshot_20220204-001108_Tachiyomi_mieruko_1](https://user-images.githubusercontent.com/72807749/152409960-ef300ee0-3911-4d0a-9bcd-d816d5b7ffa6.png) |

### But what if the author had previous works that are more famous?
Readability should win over knowing it exists on (v3) MangaDex

| Before | Now |
|--------------|-------------|
| ![Screenshot_20220204-001430_TachiyomiSY_dedede_2](https://user-images.githubusercontent.com/72807749/152409940-72e7db49-ae73-4c67-a5e8-8f18fceaa1a8.png) | ![Screenshot_20220204-001124_Tachiyomi_dedede_1](https://user-images.githubusercontent.com/72807749/152409955-ec892818-0c24-42fc-8f6f-11aecf063a56.png) |


### Or the Manga got removed from Dex
This is very rare and its fine ~~most people wont read it and will still complain~~

| Before | Now |
|--------------|-------------|
| ![Screenshot_20220204-001333_TachiyomiSY_OPM_2](https://user-images.githubusercontent.com/72807749/152409946-81a5fa89-b13e-4886-adcd-d61dc7a6220c.png) | ![Screenshot_20220204-001033_Tachiyomi_Onepiece-1](https://user-images.githubusercontent.com/72807749/152409969-770b40a1-1c5f-4909-872a-86f84cf2ae9a.png) |

### Or just became an worldwide phenomenon
Ok this was too much and now it looks way better
| Before | Now |
|--------------|-------------|
| ![Screenshot_20220204-001255_Tachiyomi_SL_2](https://user-images.githubusercontent.com/72807749/152409949-24e69658-43f7-4393-8ab4-052aa6a13005.png) | ![Screenshot_20220204-000950_Tachiyomi_SL-1](https://user-images.githubusercontent.com/72807749/152409974-0b9e742a-292c-44ae-877a-0457320f8844.png) |

alessandrojean commented:
Seems that there's some empty new lines in the end, so maybe need to clean them too?

curche commented:
Yeah, I'm figuring that one out. Looks like the whitespaces before notes section

curche commented:
Trimmed out the trailing whitespace/newlines
Removed old bbcode cleaning code
Moved regex to companion object for better performance

PR is ready
Note that however, I haven't updated build.gradle since I'll be doing a minor PR that I prefer keeping separate but not bump extVersionCode
